napari/layers/labels/labels.py

```diff
@@ -519,6 +520,10 @@ def seed_rng(self, seed_rng: Optional[int]) -> None:
     @_ImageBase.colormap.setter
     def colormap(self, colormap):
         super()._set_colormap(colormap)
+        if isinstance(self._colormap, LabelColormap):
+            self._random_colormap = self._colormap
+        else:
+            self._direct_colormap = self._colormap
```

Comment:
Will this work in general? ie, will direct colormap ever work with setting a colormap? Do we have a specific class for it like we have for LabelColormap? (maybe we should)

Reply:
You ask about `DirectLabelColormap`  from `napari.utils.colormap.colormap`?

Reply:
yes, for example. I guess what I'm saying is that I don't think people use that as an API. I also don't know whether all the right variables are set — eg. the color mode still wouldn't change, right?

Reply:
Ok, noted. Let's leave this like this in this PR, then update API (including num_colors deprecation) in a subsequent PR.